### PR TITLE
* #46: Separated pom parsing and validation.

### LINKED ISSUE #46: Separate pom file parsing and validation

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #46

### CODE REVIEW

File: src/main/java/com/exasol/releaserobot/usecases/validate/BasicMavenPomValidator.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import com.exasol.releaserobot.repository.GitRepository;
+import com.exasol.releaserobot.repository.maven.JavaMavenGitBranchContent;
+import com.exasol.releaserobot.repository.maven.MavenPom;
+import com.exasol.releaserobot.usecases.*;
+
+/**
+ * This class validates a pom file.
+ */
+public class BasicMavenPomValidator implements RepositoryValidator {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/usecases/validate/BasicMavenPomValidator.java:13]: It looks like this is not a BasicMavenPomValidator, but rather a broader concept like a MavenRepositoryValidator, which checks that a Maven repository has everything in place.

Maybe we should change the name, and also move it to its right package (not in the usecases package)

File: src/main/java/com/exasol/releaserobot/usecases/validate/ValidateInteractor.java [hunk 2]
```diff
@@ -1,56 +1,61 @@
...
     }
 
-    private Report validate(final Set<PlatformName> platformNames, final UserInput userInput) {
-        final String branch = userInput.hasGitBranch() ? userInput.getGitBranch()
-                : this.repository.getDefaultBranchName();
-        final GitRepositoryValidator repositoryValidator = new GitRepositoryValidator(this.repository);
-        final Report repositoryValidationReport = repositoryValidator.validate(branch);
-        final Report platformsValidationReport = validatePlatforms(platformNames);
+    private Report runValidation(final UserInput userInput) {
+        final Report repositoryValidationReport = validateRepositories(userInput);
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/usecases/validate/ValidateInteractor.java:37]: maybe looks more clear like:
```
Report report = new Report();
report.merge(validateRepositories(userInput));
report.merge(validatePlatforms());
report.merge(platformsValidationReport);
return report:
```

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPomParser.java [hunk 3]
```diff
@@ -1,40 +1,37 @@
...
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 
 import com.exasol.releaserobot.repository.GitRepositoryException;
 
 /**
  * Parser for {@link MavenPom}.
  */
 public class MavenPomParser {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/repository/maven/MavenPomParser.java:16]: this class should implement an interface, mainly cause if at some point you want to change, ie, from apache to another library you don't have to change the clients of this class.

For the moment it's ok, but at some point in the future this should be changed.

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java [hunk 4]
```diff
@@ -33,12 +47,43 @@ public String getVersion() {
     }
 
     /**
-     * Get deliverable name.
+     * Check if version presents.
+     *
+     * @return true if version presents
+     */
+    public boolean hasVersion() {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:54]: I see that the `getVersion` method is used in several places, but it looks like this value might not be always here. Could be any problem?

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java [hunk 5]
```diff
@@ -33,12 +47,43 @@ public String getVersion() {
...
+
+    /**
+     * Check if plugins present.
+     *
+     * @return true if plugins present
      */
-    public String getDeliverableName() {
-        return this.deliverableName;
+    public boolean hasPlugins() {
+        return this.plugins != null && !this.plugins.isEmpty();
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:86]: better avoid checking from null, and make sure the `plugins` collections is always there (even if it's empty). Probably you can do this in the builder

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java [hunk 6]
```diff
@@ -23,6 +28,15 @@ public String getArtifactId() {
         return this.artifactId;
     }
 
+    /**
+     * Check if artifact id presents.
+     * 
+     * @return true if artifactId presents
+     */
+    public boolean hasArtifactId() {
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:36]: I see that the `getArtifactId` method is used, but it looks like this value might not be always here. Could there be any problem?

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPlugin.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
+package com.exasol.releaserobot.repository.maven;
+
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+/**
+ * This class represents a Maven plugin.
+ */
+public class MavenPlugin {
+    private final String artifactId;
+    private final Xpp3Dom configuration;
```
Comment (chiaradiamarcelo) [inline src/main/java/com/exasol/releaserobot/repository/maven/MavenPlugin.java:10]: I think for now it's ok, but you should consider to create an interface for the `configuration`, if at some point in the future you use another library for it.

**Review by chiaradiamarcelo — APPROVED** [review]
Looks good, I just left some small comments.

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java [hunk 8]
```diff
@@ -33,12 +47,43 @@ public String getVersion() {
...
+    public Map<String, String> getProperties() {
+        return this.properties;
+    }
+
+    /**
+     * Check if properties present.
+     *
+     * @return true if properties present
+     */
+    public boolean hasProperties() {
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:72]: Yet. I'll need this on the next branch

File: pom.xml [hunk 9]
```diff
@@ -26,17 +26,28 @@
...
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>5.7.0</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>3.5.10</version>
+            <version>3.6.0</version>
```
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply pom.xml:44]: done

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:72]: You're right, I forgot to add this check in one place. Fixed it!

File: src/main/java/com/exasol/releaserobot/usecases/validate/BasicMavenPomValidator.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releaserobot/usecases/validate/BasicMavenPomValidator.java:13]: Renamed and moved

File: src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java (hunk 4 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/releaserobot/repository/maven/MavenPom.java:54]: Added hasVersion checks in the place I used getVersion

**Review by chiaradiamarcelo — APPROVED** [review]

**Pull request merged**